Files from eslint.seatbelt.tsv ignored on Windows

We are using eslint-seatbelt in Angular app as a team of ~20 on different OSs. 
Macos, Linux and Windows WSL users don't have any problems but Windows user without WSL set up experiencing the following problem.

Environment
```
Node version: v20.18.0
npm version: v10.8.2
Local ESLint version: v9.12.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19045
```

When he is running `pnpm eslint .`, he is seeing an output of all the errors which should be ignored according to `eslint.seatbelt.tsv`

```
D:\Projects\web\libs\vault\upload\src\lib\store\vault-upload.store.ts
   33:14  error  Class "VaultUploadStore" extends ComponentStore. Consider using signalStore instead  crexi/no-component-store
   40:51  error  Unexpected any. Specify a different type                                             @typescript-eslint/no-explicit-any
  620:88  error  Member error should be declared before member fileItem                               @typescript-eslint/member-ordering
  698:79  error  Unexpected any. Specify a different type                                             @typescript-eslint/no-explicit-any
```

And for me (MacOS)
```
/Users/dzmvasilevsky/Work/TulaCo/crexi/web/libs/vault/upload/src/lib/store/vault-upload.store.ts
   33:14  warning  Class "VaultUploadStore" extends ComponentStore. Consider using signalStore instead.
[eslint-seatbelt]: This file is temporarily allowed to have 1 error of this type.
Please tend the garden by fixing if you have the time  crexi/no-component-store
   40:51  warning  Unexpected any. Specify a different type.
[eslint-seatbelt]: This file is temporarily allowed to have 2 errors of this type.
Please tend the garden by fixing if you have the time                                            @typescript-eslint/no-explicit-any
  620:88  warning  Member error should be declared before member fileItem.
[eslint-seatbelt]: This file is temporarily allowed to have 1 error of this type.
Please tend the garden by fixing if you have the time                               @typescript-eslint/member-ordering
  698:79  warning  Unexpected any. Specify a different type.
[eslint-seatbelt]: This file is temporarily allowed to have 2 errors of this type.
Please tend the garden by fixing if you have the time                                            @typescript-eslint/no-explicit-any
```

Our first thought was that his env can't find `eslint.seatbelt.tsv` at all but we checked with
```
const file = fs.readFileSync(getRootPath('eslint.seatbelt.tsv'), "utf8");
console.log('DEBUG', file);
```
and it successfully output file contents.

So our last idea is that file pathes in `eslint.seatbelt.tsv` aren't matching the file pathes from eslint because unix / windows path notation difference. File contains unix path separator, but plugin process paths on windows machine with "\" path separator.

--- a/src/SeatbeltFile.test.ts
+++ b/src/SeatbeltFile.test.ts
@@ -131,4 +131,15 @@ describe("SeatbeltFile", () => {
     const roundtrippedFile = SeatbeltFile.fromJSON(json)
     assert.deepStrictEqual(roundtrippedFile.toJSON(), json)
   })
+
+  test("converts Windows backslashes to forward slashes", () => {
+    const windowsPath = "C:\\Users\\test\\Desktop\\eslint-seatbelt\\src\\SeatbeltFile.ts"
+    const expected = "C:/Users/test/Desktop/eslint-seatbelt/src/SeatbeltFile.ts"
+    assert.strictEqual(toPosixPath(windowsPath), expected)
+  })
+
+  test("preserves forward slashes in Unix paths", () => {
+    const unixPath = "/home/test/Desktop/eslint-seatbelt/src/SeatbeltFile.ts"
+    assert.strictEqual(toPosixPath(unixPath), unixPath)
+  })
 })

--- a/src/SeatbeltFile.ts
+++ b/src/SeatbeltFile.ts
@@ -140,7 +140,7 @@ export class SeatbeltFile {
       }
       fileState.lines.push(line)
     })
-    return new SeatbeltFile(filename, data, comments.trim())
+    return new SeatbeltFile(toPosixPath(filename), data, comments.trim())
   }
 
   static fromJSON(json: SeatbeltFileJson): SeatbeltFile {
@@ -162,7 +162,7 @@ export class SeatbeltFile {
     protected data: Map<SourceFileName, SeatbeltStateFileData>,
     public readonly comments: string = "",
   ) {
-    this.filename = path.resolve(this.filename)
+    this.filename = toPosixPath(path.resolve(this.filename))
     this.dirname = path.dirname(this.filename)
   }
 
@@ -297,14 +297,21 @@ export class SeatbeltFile {
         // Serialize maxErrors map structure if it exists, since it may have changes.
         fileState.lines = []
         fileState.maxErrors.forEach((maxErrorCount, ruleId) => {
-          fileState.lines.push({ filename, ruleId, maxErrors: maxErrorCount })
+          fileState.lines.push({ 
+            filename: toPosixPath(filename), 
+            ruleId, 
+            maxErrors: maxErrorCount 
+          })
         })
         fileState.lines.sort((a, b) =>
           a.ruleId === b.ruleId ? 0 : a.ruleId < b.ruleId ? -1 : 1,
         )
       }
       fileState.lines.forEach((line) => {
-        const encoded = (line.encoded ??= encodeLine(line))
+        const encoded = (line.encoded ??= encodeLine({
+          ...line,
+          filename: toPosixPath(line.filename)
+        }))
         lines.push(encoded)
       })
     })
@@ -380,16 +387,16 @@ export class SeatbeltFile {
 
   toRelativePath(filename: string) {
     if (!nodePath.isAbsolute(filename)) {
-      return filename
+      return toPosixPath(filename)
     }
-    return nodePath.relative(this.dirname, filename)
+    return toPosixPath(nodePath.relative(this.dirname, filename))
   }
 
   toAbsolutePath(filename: string) {
     if (nodePath.isAbsolute(filename)) {
-      return filename
+      return toPosixPath(filename)
     }
-    return nodePath.resolve(this.dirname, filename)
+    return toPosixPath(nodePath.resolve(this.dirname, filename))
   }
 }
 
@@ -399,4 +406,15 @@ function parseMaxErrors(lines: SeatbeltFileLine[]): Map<RuleId, number> {
     maxErrors.set(line.ruleId, line.maxErrors)
   })
   return maxErrors
+}
+
+/**
+ * Converts a file path to POSIX format (forward slashes).
+ * This ensures consistent path representation across different operating systems.
+ * 
+ * @param filePath - The file path to convert
+ * @returns The POSIX-style path using forward slashes
+ */
+export function toPosixPath(filePath: string): string {
+  return filePath.split(path.sep).join(path.posix.sep)
 }

